docs/wiki/EL/RLP.md

```diff
@@ -0,0 +1,198 @@
+# Recursive-Length Prefix (RLP) Serialization
+
+## Overview
+
+Recursive Length Prefix (RLP) is a core serialization protocol used within the Ethereum network to encode data. It is designed to serialize structured data, encompassing everything from transaction data to the entire state of the blockchain. This wiki page explores the internals of RLP, its encoding/decoding rules, tools available and its role in Ethereum's functionality.
+
+## Understanding Data Serialization in Ethereum
+
+![Data Serialization](/docs/images/el-architecture/data-serialization.png)
+
+_Figure: Data Serialization Flow_
+
+
+Data serialization is the process of converting data structures or objects into a byte stream for storage, transmission, or later reconstruction. In distributed systems like Ethereum, serialization is crucial for transmitting data across network nodes reliably and efficiently. While common serialization formats include JSON and XML, Ethereum uses RLP due to its simplicity and effectiveness in encoding nested arrays of bytes.
+
+## The Need for RLP in Ethereum
+
+> RLP is intended to be a highly minimalistic serialization format; its sole purpose is to store nested arrays of bytes. Unlike protobuf, BSON and other existing solutions, RLP does not attempt to define any specific data types such as booleans, floats, doubles or even integers; instead, it simply exists to store structure, in the form of nested arrays, and leaves it up to the protocol to determine the meaning of the arrays.
+> -- Ethereum's design rationale
+
+RLP is tailored for Ethereum to meet specific needs:
+- Minimalistic Design: It focuses purely on storing structure without imposing data type definitions.
+- Consistency: It guarantees byte-perfect consistency across different implementations, crucial for the deterministic nature required in blockchain operations.
+
+## RLP Encoding Rules
```

Comment:
I would swap this paragraph with the following. Charts in "How RLP Algorithm works" section are very helpful and with more context introducing the chart be more useful than heavy reading of each rule which are documented elsewhere.